todolist/pages/index.js

```diff
@@ -48,11 +48,19 @@ function Home() {
     localStorage.setItem("tasks", JSON.stringify(updatedTasks));
   };
 
-  const handleDelete = (taskId) => {
-    const updatedTasks = tasks.filter((task) => task.id !== taskId);
+ // ...
+
+const handleDelete = (taskId) => {
+  const taskIndex = tasks.findIndex((task) => task.id === taskId);
```

Comment:
Use of findIndex() vs. filter(): Currently, findIndex() is used to find the index of the task to be deleted, and then splice() is used to remove it. This approach is slightly less efficient than using filter() directly to create the updatedTasks array, as it iterates over the tasks array twice (once for finding the index and once for removing). Instead, consider using filter() directly to create the updatedTasks array in one iteration.